rust/actyx/ax/src/cmd/events/publish.rs

```diff
@@ -24,13 +27,45 @@ use util::{
 pub struct PublishOpts {
     #[structopt(flatten)]
     console_opt: ConsoleOpt,
-    /// event payload (JSON)
-    payload: serde_json::Value,
-    #[structopt(long, short)]
+
+    /// Event payload, needs to be valid JSON.
+    /// You may also pass a file in using the syntax `@file.json` or
+    /// have the command read from standard input using `@-`.
+    payload: String,
+
     /// tag (can be given multiple times)
+    #[structopt(long, short)]
     tag: Option<Vec<Tag>>,
 }
 
+fn load_json(input: String) -> ActyxOSResult<serde_json::Value> {
+    let i = if input == "@-" {
+        let stdin = std::io::stdin();
+        let mut stdin = stdin.lock(); // locking is optional
+
+        let mut line = String::new();
+        while let Ok(n_bytes) = stdin.read_to_string(&mut line) {
```

Comment:
This while loop is extraneous: `read_to_string` already reads to the end of input by itself.